src/main/java/seedu/address/logic/relationship/DeleteRelationshipCommand.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.logic.relationship;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import seedu.address.model.person.Person;
+import seedu.address.model.person.relationship.Relationship;
+import seedu.address.model.person.relationship.RelationshipManager;
+
+public class DeleteRelationshipCommand {
+    private final Map<String, Person> personMap;
+    private final RelationshipManager relationshipManager;
+
+    /**
+     * Creates a new DeleteRelationshipCommand with the given person map and relationship manager.
+     *
+     * @param personMap           A map linking UUID strings to Person objects.
+     * @param relationshipManager The RelationshipManager instance managing the relationships.
+     */
+    public DeleteRelationshipCommand(Map<String, Person> personMap, RelationshipManager relationshipManager) {
+        this.personMap = personMap;
+        this.relationshipManager = relationshipManager;
+    }
+
+    /**
+     * Parses and executes the given delete relationship command.
+     *
+     * @param command The delete relationship command to parse and execute.
+     * @throws IllegalArgumentException If the command is invalid.
+     */
+    public void parseCommand(String command) {
+        String[] commandParts = command.split("\\s+");
+
+        if (commandParts.length != 4) {
+            throw new IllegalArgumentException("Invalid command format");
+        }
+
+        String relationType = commandParts[1].substring(1);
+        String[] personUuids = commandParts[2].split(",");
+        if (personUuids.length != 2) {
+            throw new IllegalArgumentException("Invalid command format");
+        }
+
+        String uuid1 = matchUuid(personUuids[0]);
+        String uuid2 = matchUuid(personUuids[1]);
+
+        List<Relationship> relationships = relationshipManager.getRelationships(relationType);
+        if (relationships.isEmpty()) {
+            throw new IllegalArgumentException("No relationship found.");
+        }
+
+        // Check if the relationship exists and remove it
+        boolean relationshipFound = false;
+        for (Relationship relationship : relationships) {
+            UUID person1Uuid = relationship.getPerson1();
+            UUID person2Uuid = relationship.getPerson2();
+            if ((person1Uuid.equals(UUID.fromString(uuid1)) && person2Uuid.equals(UUID.fromString(uuid2))) ||
+                    (person1Uuid.equals(UUID.fromString(uuid2)) && person2Uuid.equals(UUID.fromString(uuid1)))) {
+                relationshipManager.deleteRelationship(relationType, relationship);
```

Comment:
Here you can refactor the code to directly throw an exception if the relationship is found and deleted, simplifying the logic a little.

Reply:
i think it may be cleaner if u implemented the logic in the relationship class. Implement this equals method in the relationship class so that next time u can use it again in editRelationship etc